src/Shared/CertificateGeneration/CertificateManager.cs

```diff
@@ -117,6 +122,25 @@ internal class CertificateManager
             bool HasOid(X509Certificate2 certificate, string oid) =>
                 certificate.Extensions.OfType<X509Extension>()
                     .Any(e => string.Equals(oid, e.Oid.Value, StringComparison.Ordinal));
+
+            bool MatchesVersion(X509Certificate2 c)
+            {
+                var byteArray = c.Extensions.OfType<X509Extension>()
+                    .Where(e => string.Equals(AspNetHttpsOid, e.Oid.Value, StringComparison.Ordinal))
+                    .Single()
+                    .RawData;
+
+                if (byteArray.Length == AspNetHttpsOidFriendlyName.Length || byteArray.Length == 0)
```

Comment:
Let's sanity check the first byte or something as well here (`byteArray[0] == (byte)'A'`). Just in case we happen to add more fields and the length ends up matching.

Reply:
```suggestion
                if ((byteArray.Length == AspNetHttpsOidFriendlyName.Length && byteArray[0] == (byte)'A') || byteArray.Length == 0)
```